# Logistic Regression Binomial
### Introduction of Dataset

The name of the dataset is Haberman's dataset. it is used for the cancer survival prdiction. At Billing Hospital, University of Chicago created the dataset based on 306 female breast cancer patient from 1958 to 2006. 
The identity of various attributes of Haberman dataset.

* Axil Node: Number of lymph nodes or lymph glands surrounding the tumor
* Age: The ageof the patient.
* Operational Year: The year of surgery took place.
* Survival Status: Those who survived more than 5 years after the surgery are given status 1 and those who died before 5 years   are given status 2.


#### Load the necessary packages and the dataset

In [20]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.mlab as mlab
import io

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix, accuracy_score
from sklearn.metrics import roc_auc_score
from sklearn.metrics import roc_curve

df = pd.read_csv('data-heberman.csv')
df.head()

age  operation_year  axil_nodes  status
0     30              64           1       1
1     30              62           3       1
2     30              65           0       1
3     31              59           2       1
4     31              65           4       1
..   ...             ...         ...     ...
301   75              62           1       1
302   76              67           0       1
303   77              65           3       1
304   78              65           1       2
305   83              58           2       2

[306 rows x 4 columns]

In [24]:
X = df[['age','operation_year','axil_nodes']]
y = df['status'].map({1:0, 2:1})

In [25]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=5)


##### Fit the model

In [14]:
model = LogisticRegression()
model.fit(x_train, y_train)


LogisticRegression()

##### Prediction

In [16]:
predictions = model.predict(x_test)
predictions

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int64)

#### Check the accuracy of the model

In [19]:
print('Coefficent:', model.coef_)
print('Intercept:', model.intercept_)
print('Classification Report: \n', classification_report(y_test, predictions))
print('Confusion Metrix:\n', confusion_matrix(y_test, predictions))
print('Accuracy Score:', accuracy_score(y_test, predictions))
print('Precision:', metrics.precision_score(y_test, predictions))
print('Recall:', metrics.recall_score(y_test, predictions))

Coefficent: [[ 0.01558675 -0.03293527  0.07053309]]
Intercept: [-0.00057878]
Classification Report: 
               precision    recall  f1-score   support

           0       0.81      1.00      0.90        60
           1       1.00      0.18      0.30        17

    accuracy                           0.82        77
   macro avg       0.91      0.59      0.60        77
weighted avg       0.85      0.82      0.76        77

Confusion Metrix:
 [[60  0]
 [14  3]]
Accuracy Score: 0.8181818181818182
Precision: 1.0
Recall: 0.17647058823529413


#### We see that our model can provide about 82% accurate prediction

#### Prediction

###### Suppose a patient is 35 years old, had 15 lymph nodes next to his cancer cell and done surgery in 1964, now we have to predict using the model, did he live more tahn 5 years or did he die before that?

In [21]:
new_observation = [[35, 64, 15]]

model.predict(new_observation)

array([0], dtype=int64)

#### So he survived more than 5 years

#### Probability value of the classes for this perticular patient

In [23]:
model.predict_proba(new_observation)

array([[0.62361094, 0.37638906]])

#### SO we see that this particular patient has probability to survive more than 5 years is 62% and not survive 5 than more 5 years is 37%